### DISCUSSION TITLE
Failed To Create User

### DISCUSSION THREAD

**Discussion by sveltarne** [pocketbase/pocketbase#855] [Q&A]
Steps to reproduce:

1. Send a post request to pocketbaseurl/api/users with the data
```{"email":"diamoncyt@gmail.com","password":"12345678","passwordConfirm":"12345678"}```
2. You'll get a response saying
`{
	"code": 400,
	"message": "Failed to create user.",
	"data": {}
} `
![image](https://user-images.githubusercontent.com/114943151/198362552-55411469-aaca-425a-9904-7be4eab836e1.png)

I've tried doing this request from a application instead of a api client and it also doesnt work. The api client i am using is insomnia, and i am using node + sveltekit's fetch in my application

**Comment by ganigeorgiev** [comment]
@sveltarne I cannot reproduce the issue. My guess is that your request doesn't have `content-type: application/json` header.
For the client-side, if you are using Node 17+ make sure that you are sending the request to `127.0.0.1:8090` instead of `localhost:8090`, because in newer node versions `localhost` resolves to the IPv6 address by default, where the Go server will bind to the IPv4 address.

**Reply by ganigeorgiev** [reply]
_btw. You can also start the PocketBase executable with `--debug` for more detailed errors and/or check the logs from the "Admin UI > Logs"._

**Reply by sveltarne** [reply]
> My guess is that your request doesn't have content-type: application/json header.  

If you [take a look at the js file](https://github.com/sveltarne/observer/blob/master/src/lib/scripts/req.js) im using to make requests on my web app, you can see that the header is there.

> make sure that you are sending the request to 127.0.0.1:8090 instead of localhost:8090

I've changed it to that, and will keep it like that, but that didn't resolve the error.

Here are the logs on the admin ui:
![image](https://user-images.githubusercontent.com/114943151/198409255-b96f385a-6bbf-4bd4-9341-0175a71186bd.png)

And whenever i make the request, it logs on the executive:
```[0.00ms] SELECT count(*) FROM `_users` WHERE (NOT (`id`='')) AND (`email`='diamoncyt@gmail.com') LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE (`id`='profiles') OR (`name`='profiles') LIMIT 1
[0.00ms] INSERT INTO `_users` (`created`, `email`, `id`, `lastResetSentAt`, `lastVerificationSentAt`, `passwordHash`, `tokenKey`, `updated`, `verified`) VALUES ('2022-10-27 22:26:45.962', 'diamoncyt@gmail.com', '330344eeu1hq50u', '', '', '$2a$13$4IHUTnoaHd7R.yomTxeebuN37dgKjdDioOhmU79xtdt4S/p8fISyC', '4C5WROuXQ72flMwZGQFOwndGTloJiZGTbqNpHmd1ktulFaE1ed', '2022-10-27 22:26:45.962', false)
2022/10/27 17:26:45 attempt to write a readonly database (8)
2022/10/27 17:27:05 Log save failed: attempt to write a readonly database (8)
2022/10/27 17:27:05 Logs delete failed: attempt to write a readonly database (8)```

**Reply by ganigeorgiev** [reply]
Hm, that's strange. It seems that the `pb_data` dir permissions are not ok. Maybe is an antivirus/firewall software that has changed them?

In any case, make sure that the OS user that is running the PocketBase executable has permissions to read/modify the `pb_data` directory and files.

**Reply by ganigeorgiev** [reply]
@sveltarne Another possibility for the `attempt to write a readonly database` could be also if you've started initially the executable with one user (eg. root/administrator) and then later with a different one because the db files (`pb_data/data.db` and `pb_data/logs.db`) will be created with 644 by default, meaning that only the owner will have read and write access, while the other users only read.

If after fixing the permissions you are still getting the same error, you could try testing with a different `pb_data` directory using the `--dir` flag (it will be created automatically by the process if missing):

```sh
./pocketbase serve --dir="./pb_data2" --debug
```
